htdocs/commande/class/api_orders.class.php

```diff
@@ -230,6 +235,10 @@ public function index($sortfield = "t.rowid", $sortorder = 'ASC', $limit = 100,
 				if ($commande_static->fetch($obj->rowid)) {
 					// Add external contacts ids
 					$commande_static->contacts_ids = $commande_static->liste_contact(-1, 'external', 1);
+					// Add online_payment_url, cf #20477
+					require_once DOL_DOCUMENT_ROOT.'/core/lib/payments.lib.php';
+					$commande_static->online_payment_url = getOnlinePaymentUrl(0, 'order', $commande_static->ref);
```

Comment:
If you add this property, you must also declare the property into the file commande.class.php

Reply:
@eldy hmmm ok ... but I'm afraid it does not cause bugs in other parts that would be based on the list of properties to do a treatment or display, right? 

For example in that sort of code:

```
foreach ($object->fields as $key => $val) {
```

So i don't know if there are other parts of code to check before commit commande.class.php ?

Reply:
I was just speaking about adding:
public $online_payment_url;
into top of header. No need to ass it into ->fields.